### PR TITLE
Require config

### PR DESCRIPTION
**Pull request by perolavsvendsen**
Require that config is provided, and that it contains specific keys. The list of specific keys used here are collected based on usage in existing code.

### CODE REVIEW

File: src/fmu/dataio/dataio.py [hunk 1]
```diff
@@ -374,29 +370,85 @@ def pwd(self):
...
+
+        Returns bool. True if config is valid.
+        """
+
+        config_required_keys = ["access", "masterdata", "model"]
+
+        for required_key in config_required_keys:
+            if required_key not in config:
+                raise ValueError("Required key '%s' not found in config.", required_key)
+
```
Comment (jcrivenaes) [inline src/fmu/dataio/dataio.py:449]: Use f formatting, not logging style (%s):
`raise ValueError(f"Required key '{required_key}' not found in config.")`

File: src/fmu/dataio/dataio.py [hunk 2]
```diff
@@ -374,29 +370,85 @@ def pwd(self):
...
                 raise
 
+        return config
+
+    def _config_validate(self, config):
+        """Validate the config.
+
+        Check that required keys in config are present.
+
+        Returns bool. True if config is valid.
```
Comment (jcrivenaes) [inline src/fmu/dataio/dataio.py:441]: Mention that this currently is not a full validation; it only tests that some first level keys are present

**Review by jcrivenaes — COMMENTED** [review]
Nice, look at comments and remove the Draft status

File: src/fmu/dataio/dataio.py (hunk 2 above)
Comment (perolavsvendsen) [reply src/fmu/dataio/dataio.py:441]: I also noticed that the docstring says "return bool", while the function is in fact returning the config. The boolean value of the config - if it makes it this far - is True, so it works. But I will have a look at this and see if it should in fact be returning a boolean or if the docstring is wrong.

**Review by jcrivenaes — APPROVED** [review]
👍

**Pull request merged**